rocketmq/src/main/java/org/apache/seata/integration/rocketmq/TCCRocketMQImpl.java

```diff
@@ -0,0 +1,89 @@
+import org.apache.rocketmq.common.message.Message;
+import org.apache.rocketmq.remoting.exception.RemotingException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * the type TCCRocketMQImpl
+ */
+public class TCCRocketMQImpl implements TCCRocketMQ {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TCCRocketMQImpl.class);
+
+    private static final String ROCKET_MSG_KEY = "ROCKET_MSG";
+    private static final String ROCKET_SEND_RESULT_KEY = "ROCKET_SEND_RESULT";
+
+    private SeataMQProducer producer;
+
+    @Override
+    public void setProducer(SeataMQProducer producer) {
+        this.producer = producer;
+    }
+
+    @Override
+    public SendResult prepare(Message message, long timeout) throws MQClientException {
+        BusinessActionContext context = BusinessActionContextUtil.getContext();
+        LOGGER.info("RocketMQ message send prepare, xid = {}", context.getXid());
+        Map<String, Object> params = new HashMap<>(8);
+        SendResult sendResult = producer.doSendMessageInTransaction(message, timeout, context.getXid(), context.getBranchId());
+        params.put(ROCKET_MSG_KEY, message);
+        params.put(ROCKET_SEND_RESULT_KEY, sendResult);
+        BusinessActionContextUtil.addContext(params);
+        return sendResult;
+    }
+
+    @Override
+    public boolean commit(BusinessActionContext context)
+            throws UnknownHostException, MQBrokerException, RemotingException, InterruptedException, TimeoutException, TransactionException {
+        Message message = context.getActionContext(ROCKET_MSG_KEY, Message.class);
+        SendResult sendResult = context.getActionContext(ROCKET_SEND_RESULT_KEY, SendResult.class);
+        if (message == null || sendResult == null) {
+            throw new TransactionException("TCCRocketMQ commit but cannot find message and sendResult");
+        }
+        producer.getDefaultMQProducerImpl().endTransaction(message, sendResult, LocalTransactionState.COMMIT_MESSAGE, null);
+        LOGGER.info("RocketMQ message send commit, xid = {}, branchId = {}", context.getXid(), context.getBranchId());
+        return true;
+    }
+
+    @Override
+    public boolean rollback(BusinessActionContext context)
+            throws UnknownHostException, MQBrokerException, RemotingException, InterruptedException, TransactionException {
+        Message message = context.getActionContext(ROCKET_MSG_KEY, Message.class);
+        SendResult sendResult = context.getActionContext(ROCKET_SEND_RESULT_KEY, SendResult.class);
+        if (message == null || sendResult == null) {
+            LOGGER.error("TCCRocketMQ rollback but cannot find message and sendResult");
+        }
+        producer.getDefaultMQProducerImpl().endTransaction(message, sendResult, LocalTransactionState.ROLLBACK_MESSAGE, null);
```

Comment:
非常多的代码使用了getDefaultMQProducerImpl，应该将其设置为成员变量，避免无意义开销
A very large amount of code uses getDefaultMQProducerImpl, which should be set as a member variable to avoid pointless overhead